### ISSUE TITLE
False positive on `isinstance(dict(a=0), dict)` in 1.1.310

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5176]
**Describe the bug**
As the title.

**To Reproduce**
See sample.

**Expected behavior**
No error is reported.

**Screenshots or Code**
```python
_ = isinstance(dict(a=0), dict)
```
![image](https://github.com/microsoft/pyright/assets/15389744/f8861cef-1500-406a-9104-16b620551d6d)

**VS Code extension or command-line**
Both.

**Additional context**
N/A

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.311, which I just published. It will also be included in a future release of pylance.
